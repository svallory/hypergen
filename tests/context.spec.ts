import { describe, it, expect } from 'bun:test'
import context from '~/context.js'

describe('context', () => {
  it('should populate with capitalized keys', () => {
    const ctx = context({ name: 'foobar' })
    delete ctx.h.path // platform dependent
    expect(ctx).toMatchSnapshot()
  })
  it('allows helpers to be initialized with current context', () => {
    const helpers = (args, ctx) => ({
      testArgs: () => args,
      testCtx: () => ctx,
    })
    const locals = {}
    const config = {
      helpers,
    }
    const ctx = context(locals, config)
    expect(typeof ctx.h.testArgs).toBe('function')
    expect(typeof ctx.h.testCtx).toBe('function')
    expect(ctx.h.testArgs()).toBe(locals)
    expect(ctx.h.testCtx()).toBe(config)
  })
})
