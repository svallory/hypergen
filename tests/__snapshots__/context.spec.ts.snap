// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`context should populate with capitalized keys 1`] = `
{
  "Name": "Foobar",
  "Names": "Foobars",
  "h": {
    "capitalize": [Function: capitalize],
    "changeCase": {
      "camel": [Function],
      "camelCase": [Function],
      "constant": [Function],
      "constantCase": [Function],
      "dot": [Function],
      "dotCase": [Function],
      "header": [Function],
      "headerCase": [Function],
      "hyphen": [Function],
      "hyphenCase": [Function],
      "isLower": [Function],
      "isLowerCase": [Function],
      "isUpper": [Function],
      "isUpperCase": [Function],
      "kebab": [Function],
      "kebabCase": [Function],
      "lcFirst": [Function],
      "lower": [Function],
      "lowerCase": [Function],
      "lowerCaseFirst": [Function],
      "no": [Function],
      "noCase": [Function],
      "param": [Function],
      "paramCase": [Function],
      "pascal": [Function],
      "pascalCase": [Function],
      "path": [Function],
      "pathCase": [Function],
      "sentence": [Function],
      "sentenceCase": [Function],
      "snake": [Function],
      "snakeCase": [Function],
      "swap": [Function],
      "swapCase": [Function],
      "title": [Function],
      "titleCase": [Function],
      "ucFirst": [Function],
      "upper": [Function],
      "upperCase": [Function],
      "upperCaseFirst": [Function],
    },
    "inflection": {
      "_apply_rules": [Function],
      "camelize": [Function],
      "capitalize": [Function],
      "classify": [Function],
      "dasherize": [Function],
      "demodulize": [Function],
      "foreign_key": [Function],
      "humanize": [Function],
      "indexOf": [Function],
      "inflect": [Function],
      "ordinalize": [Function],
      "pluralize": [Function],
      "singularize": [Function],
      "tableize": [Function],
      "titleize": [Function],
      "transform": [Function],
      "undasherize": [Function],
      "underscore": [Function],
      "version": "1.13.1",
    },
  },
  "name": "foobar",
  "names": "foobars",
}
`;
